src/charm.py

```diff
@@ -206,6 +219,85 @@ def _on_peer_relation_changed(self, event: RelationChangedEvent) -> None:
         ):
             self.unit.status = ActiveStatus()
 
+    # flake8: noqa: C901
+    def _on_db_router_relation_changed(self, event: RelationChangedEvent) -> None:
+        """Handle the db_router relation changed event."""
+        if not self.unit.is_leader():
+            return
+
+        self.unit.status = MaintenanceStatus("Setting up db-router relation")
+
+        # Get the application data from the relation databag
+        # Abstracted is the fact that it is received from the leader unit of the related app
+        unit_address = str(self.model.get_binding(PEER).network.bind_address)
+        unit_names = " ".join([unit.name for unit in event.relation.units])
+
+        event_relation_databag = event.relation.data[self.unit]
+        event_relation_databag["db_host"] = json.dumps(unit_address)
+
+        application_unit = list(event.relation.units)[0]
+        application_data = copy.deepcopy(event.relation.data[application_unit])
```

Comment:
Why the need for deepcopy here? Any race condition identified?

Reply:
no need anymore, i was previously deleting a key from the application_data, but no more. removed in 5a2d6a26ef82b30eb9937885ae5460ab5f22024f